tests/gravatar-parquet/src/mapping/gravatar.ts

```diff
@@ -0,0 +1,127 @@
+import {CommonHandlerContext, EvmBlock} from '@subsquid/evm-processor'
+import {LogItem, TransactionItem} from '@subsquid/evm-processor/lib/interfaces/dataSelection'
+import {Store} from '../db'
+import * as spec from '../abi/0x2e645469f354bb4f5c8a05b3b30a929361cf77ec'
+import {normalize} from '../util'
+
+export {spec}
+
+export const address = '0x2e645469f354bb4f5c8a05b3b30a929361cf77ec'
+
+type EventItem = LogItem<{evmLog: {topics: true, data: true}, transaction: {hash: true}}>
+type FunctionItem = TransactionItem<{transaction: {hash: true, input: true, value: true}}>
+
+export function parse(ctx: CommonHandlerContext<Store>, block: EvmBlock, item: EventItem | FunctionItem) {
+    switch (item.kind) {
+        case 'evmLog':
+            return parseEvent(ctx, block, item)
+        case 'transaction':
+            return parseFunction(ctx, block, item)
+    }
+}
+
+function parseEvent(ctx: CommonHandlerContext<Store>, block: EvmBlock, item: EventItem) {
+    try {
+        switch (item.evmLog.topics[0]) {
+            case spec.events['NewGravatar'].topic: {
+                let e = normalize(spec.events['NewGravatar'].decode(item.evmLog))
+                ctx.store.GravatarEventNewGravatar.write({
+                    id: item.evmLog.id,
+                    blockNumber: block.height,
+                    blockTimestamp: new Date(block.timestamp),
+                    transactionHash: item.transaction.hash,
+                    contract: item.address,
+                    eventName: 'NewGravatar',
+                    id0: e[0],
+                    owner: e[1],
+                    displayName: e[2],
+                    imageUrl: e[3],
+                })
+            }
+            case spec.events['UpdatedGravatar'].topic: {
+                let e = normalize(spec.events['UpdatedGravatar'].decode(item.evmLog))
+                ctx.store.GravatarEventUpdatedGravatar.write({
+                    id: item.evmLog.id,
+                    blockNumber: block.height,
+                    blockTimestamp: new Date(block.timestamp),
+                    transactionHash: item.transaction.hash,
+                    contract: item.address,
+                    eventName: 'UpdatedGravatar',
+                    id0: e[0],
+                    owner: e[1],
+                    displayName: e[2],
+                    imageUrl: e[3],
+                })
+            }
+        }
+    }
+    catch (error) {
+        ctx.log.error({error, blockNumber: block.height, blockHash: block.hash, address}, `Unable to decode event "${item.evmLog.topics[0]}"`)
+    }
+}
+
+function parseFunction(ctx: CommonHandlerContext<unknown>, block: EvmBlock, item: FunctionItem) {
```

Comment:
Generated code fails to compile with `'ctx.store' is of type 'unknown'`. I replaced `CommonHandlerContext<unknown>` -> `CommonHandlerContext<any>` and it went through, but there may be a better way.

Reply:
I've already fixed it, haven't I?

Reply:
Seems so